global_costmap:
  plugins: 
    - {name: static_layer, type: "costmap_2d::StaticLayer"}
    - {name: inflation_layer, type: "costmap_2d::InflationLayer"}
  global_frame: map
  robot_base_frame: base_link
  footprint: [[-0.07, -0.22], [-0.07, 0.22], [0.43, 0.22], [0.43, -0.22]]
  update_frequency: 2.0
  publish_frequency: 2.0
  static_layer:
    map_topic: "map"
    track_unknown_space: false
    subscribe_to_updates: true
  inflation_layer:
    inflation_radius: 0.50
    cost_scaling_factor: 3.0

local_costmap:
  plugins: 
    - {name: obstacle_layer, type: 'costmap_2d::ObstacleLayer'}
    # - {name: sonar_layer,   type: "range_sensor_layer::RangeSensorLayer"}
    - {name: inflation_layer, type: "costmap_2d::InflationLayer"}
  global_frame: odom
  robot_base_frame: base_link
  footprint: [[-0.07, -0.22], [-0.07, 0.22], [0.43, 0.22], [0.43, -0.22]]
  rolling_window: true
  width: 4.0
  height: 4.0
  resolution: 0.05
  update_frequency: 2.0
  publish_frequency: 2.0

  obstacle_layer:
    observation_sources: laser_scan_sensor
    laser_scan_sensor: 
      sensor_frame: base_laser
      topic: /scan
      data_type: LaserScan
      marking: true
      clearing: true
  # sonar_layer:
  #   topics: [ /dtw_robot1/sonar_back/range ]
  #   clear_on_max_reading: true
  inflation_layer:
    inflation_radius: 0.20
    cost_scaling_factor: 10.0

local_planner:
  sim_time: 3.0
  max_vel_x: 0.15
  min_vel_x: 0.05
  max_vel_theta: 0.3
  min_in_place_vel_theta: 0.05
  max_rotational_vel: 0.3
  acc_lim_theta: 3.2
  acc_lim_x: 2.5
  holonomic_robot: false
  meter_scoring: true
  controller_frequency: 10.0
  yaw_goal_tolerance: 3.0
  pdist_scale : 0.6 
  gdist_scale : 0.2
  occdist_scale: 0.1